File: lab3/msLab3/src/classes/Route.ts
import { Element } from "./Element.js";
import { Job } from "./Job.js";

export class Route {
  private readonly element: Element | null;
  private priority: number = 0;
  private probability: number = 1.0;
  private block: ((job: Job) => boolean) | null = null;

  constructor(
    element: Element | null,
    probability: number = 1.0,
    priority: number = 0,
    block?: (job: Job) => boolean
  ) {
    this.element = element;
    this.probability = probability;
    this.priority = priority;
    if (block) {
      this.block = block;
    }
  }

  public isBlocked(job: Job): boolean {
    if (this.block === null) {
      return false;
    }
    try {
      return this.block(job);
    } catch (error) {
      throw new Error(`${error}`);
    }
  }

  public getElement(): Element | null {
    return this.element;
  }

  public getPriority(): number {
    return this.priority;
  }

  public getProbability(): number {
    return this.probability;
  }

  public setBlock(block: (job: Job) => boolean): void {
    this.block = block;
  }
}

File: lab1/pythonProject/core/core/Route.py
class Route:
    def __init__(self, element, probability=1.0, priority=0, block=None):
        self.element = element
        self.probability = probability
        self.priority = priority
        self.block = block

    def is_blocked(self, job):
        if self.block is None:
            return False
        try:
            return self.block(job)
        except Exception as e:
            raise RuntimeError(e)

    def get_element(self):
        return self.element

    def get_priority(self):
        return self.priority

    def get_probability(self):
        return self.probability

    def set_block(self, block):
        self.block = block